OpenRA.Mods.Common/Traits/SupportPowers/SupportPowerManager.cs

```diff
@@ -243,6 +253,14 @@ public void Activate(Order order)
 			if (power == null)
 				return;
 
+			if (!HasSufficientFunds(power))
+				return;
+
+			if (power.Info.Cost > 0)
+				manager.PlayerResources.TakeCash(power.Info.Cost);
+			else
+				manager.PlayerResources.GiveCash(-power.Info.Cost);
```

Comment:
IMO better not to call GiveCash for Cost==0, currently no effect, but there could be lately.

Reply:
We now have a method that combines both Take/Give cash so this split can go away.

Reply:
Can't use ChangeCash here - this should be an all-or-nothing case but that method adjusts the amount taken to always pass.